File: plot/src/main/java/net/fwitz/math/plot/binary/dual/DualNumberFunctionPlot.java
package net.fwitz.math.plot.binary.dual;

import net.fwitz.math.binary.dual.DualNumber;
import net.fwitz.math.plot.binary.BinaryNumberFunctionPlot;
import net.fwitz.math.plot.binary.dual.color.DomainColoringDual;

public class DualNumberFunctionPlot extends BinaryNumberFunctionPlot<
        DualNumberFunctionPlot,
        DualNumberFunctionPanel,
        DualNumber,
        DualNumber> {

    public DualNumberFunctionPlot(String title) {
        super(title);
        colorFn(new DomainColoringDual());
    }

    @Override
    public DualNumber value(double x, double y) {
        return DualNumber.dualNumber(x, y);
    }

    @Override
    protected DualNumberFunctionPanel createPanel() {
        return new DualNumberFunctionPanel(this);
    }
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/dual/DualNumberFunctionPlot.kt
package net.fwitz.math.plot.binary.dual

import net.fwitz.math.binary.dual.DualNumber
import net.fwitz.math.plot.binary.dual.color.DomainColoringDual
import net.fwitz.math.plot.binary.BinaryNumberFunctionPlot

class DualNumberFunctionPlot(title: String) : BinaryNumberFunctionPlot<
        DualNumberFunctionPlot,
        DualNumberFunctionPanel,
        DualNumber,
        DualNumber
>(title) {
    init {
        colorFn(DomainColoringDual)
    }

    override fun value(x: Double, y: Double) = DualNumber(x, y)

    override fun createPanel() = DualNumberFunctionPanel(this)
}